### PR TITLE
Add a note to install libgtk before compiling on Linux

### PR DESCRIPTION
**Pull request by YassinEldeeb**
when you try compiling any crate with `rfd` as a dependency without installing `libgtk` on linux, It'll always fail with this hard to debug error message, so better add a note here to aware people before compiling to linux.

This is also noted here in [`tao`](https://github.com/tauri-apps/tao#platform-specific-notes)

```sh
error: failed to run custom build command for `atk-sys v0.15.1`

Caused by:
  process didn't exit successfully: `/home/runner/work/deadliner/deadliner/target/debug/build/atk-sys-4862881347b4c8db/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=ATK_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`"pkg-config" "--libs" "--cflags" "atk" "atk >= 2.18"` did not exit successfully: exit status: 1
  error: could not find system library 'atk' required by the 'atk-sys' crate

  --- stderr
  Package atk was not found in the pkg-config search path.
  Perhaps you should add the directory containing `atk.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'atk' found
  Package atk was not found in the pkg-config search path.
  Perhaps you should add the directory containing `atk.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'atk' found

warning: build failed, waiting for other jobs to finish...
error: build failed
```

### CODE REVIEW

**Comment by PolyMeilex** [comment]
Look like dependencies section got removed in #43, I missed that.
Could we add a note along the lines of:
> Also if you want to avoid GTK dependency you can also read [xdg-portal section](https://docs.rs/rfd/latest/rfd/#xdg-desktop-portal-backend) of RFD docs

**Comment by Be-ing** [comment]
This is already documented: https://docs.rs/rfd/latest/rfd/#gtk-backend

**Comment by Be-ing** [comment]
I still recommend to make `xdg-portal` the default to avoid developers of downstream applications stumbling over this.

**Comment by Be-ing** [comment]
I don't think repeating documentation in multiple places is a great idea.

**Comment by PolyMeilex** [comment]
So let's leave the link to [gtk-backend](https://docs.rs/rfd/latest/rfd/#gtk-backend) section of the docs as a dependency note in the README, no duplication and better discoverability.

**Comment by Be-ing** [comment]
Or perhaps this:

"Refer to [Linux & BSD backends](https://docs.rs/rfd/latest/rfd/#linux--bsd-backends) for information about dependencies."

**Comment by YassinEldeeb** [comment]
So what are we gonna do?

**Comment by PolyMeilex** [comment]
Sure I agree. 

If you want to add commands, just plop a new table column in the docs here: https://github.com/PolyMeilex/rfd/blob/b756bb91f3ed1a5112e00f86671a578ecc544d31/src/lib.rs#L50

Personal I would omit the `sudo` prefix to make things shorter, I expect people to know their package manager enough to know that, but feel free to do whatever you think is the best.

File: README.md [hunk 1]
```diff
@@ -13,3 +13,20 @@ It provides both asynchronous and synchronous APIs. Supported platforms:
...
+### Arch Linux / Manjaro:
+
+```sh
+sudo pacman -S gtk3
+```
+
+### Debian / Ubuntu:
+```sh
+sudo apt install libgtk-3-dev
+```
```
Comment (YassinEldeeb) [inline README.md:32]: So first of all for people who will land on the readme.
```suggestion
Please refer to [Linux & BSD backends](https://docs.rs/rfd/latest/rfd/#linux--bsd-backends) for information about the needed dependencies to be able to compile on Linux.
```
```

File: src/lib.rs [hunk 2]
```diff
@@ -35,6 +35,8 @@
 //! };
 //! ```
 //!
+//! # Important Notes before compiling!
+//!
```
Comment (YassinEldeeb) [inline src/lib.rs:39]: So I've put a header for people like me who rushes through the documentation to aware them of a few notes before compiling on Linux.

File: src/lib.rs [hunk 3]
```diff
@@ -47,11 +49,11 @@
...
 //!
-//! | Distribution    | Package name |
+//! | Distribution    | Installation Command |
 //! | --------------- | ------------ |
-//! | Fedora          | gtk3-devel   |
-//! | Arch            | gtk3         |
-//! | Debian & Ubuntu | libgtk-3-dev |
+//! | Fedora          | apt install gtk3-devel   |
+//! | Arch            | apt install gtk3         |
+//! | Debian & Ubuntu | apt install libgtk-3-dev |
```
Comment (YassinEldeeb) [inline src/lib.rs]: And here I've just edited the table column to Installation command instead of package name to be a bit clearer.

**Comment by YassinEldeeb** [comment]
Feel free to correct my changes If I messed up anything, I only have 15 minutes of Linux experience, lol.

File: src/lib.rs [hunk 4]
```diff
@@ -47,11 +49,11 @@
 //! backend requires the C library and development headers to be installed to build RFD. The package
 //! names on various distributions are:
 //!
-//! | Distribution    | Package name |
+//! | Distribution    | Installation Command |
 //! | --------------- | ------------ |
-//! | Fedora          | gtk3-devel   |
-//! | Arch            | gtk3         |
-//! | Debian & Ubuntu | libgtk-3-dev |
+//! | Fedora          | apt install gtk3-devel   |
```
Comment (Be-ing) [inline src/lib.rs:54]:
```suggestion
//! | Fedora          | dnf install gtk3-devel   |
```

File: src/lib.rs [hunk 5]
```diff
@@ -47,11 +49,11 @@
...
+//! | Arch            | apt install gtk3         |
```
Comment (Be-ing) [inline src/lib.rs:55]:
```suggestion
//! | Arch            | pacman -S gtk3         |
```

File: src/lib.rs (hunk 2 above)
Comment (Be-ing) [inline src/lib.rs:39]: I don't think this adds anything. To the contrary, it creates a section header with no content.

File: README.md (hunk 1 above)
Comment (Be-ing) [inline README.md:32]: Please replace with a link to the documentation as discussed.

File: src/lib.rs (hunk 2 above)
Comment (YassinEldeeb) [reply src/lib.rs:39]:
```suggestion
```
Okay, better remove it.

File: README.md (hunk 1 above)
Comment (YassinEldeeb) [reply README.md]: done!

**Review by YassinEldeeb — COMMENTED** [review]
I think It's now ready to be merged.

**Review by YassinEldeeb — COMMENTED** [review]
LGTM.

**Review by PolyMeilex — APPROVED** [review]
LGTM

**Pull request merged**